src/Spectre.libClassifier/DownsampledOpenCVDataset.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+* DownsampledOpenCVDataset.cpp
+* Structure having same amount of cancer and noncancer cells for fitness function random multiple validation
+*
+Copyright 2017 Spectre Team
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "DownsampledOpenCVDataset.h"
+#include "RandomSplitter.h"
+#include "ObservationExtractor.h"
+#include "Spectre.libGenetic/DataTypes.h"
+
+namespace spectre::supervised
+{
+DownsampledOpenCVDataset::DownsampledOpenCVDataset(OpenCvDataset data, size_t maximumSubsetSize, double trainingRate)
+    : m_TrainingRate(trainingRate)
+{
+    std::vector<DataType> cancerData{};
```

Comment:
If we are at some point investigating drug distribution or any other optimization problem for which GA is used, what is the meaning of *cancer data* in such context?

Reply:
done